stdlib/src/collections/counter.mojo

```diff
@@ -0,0 +1,355 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+
+from collections.dict import Dict, _DictKeyIter, _DictValueIter, _DictEntryIter
+
+
+struct Counter[V: KeyElement](Sized, CollectionElement, Boolable):
+    """A container for counting hashable items.
+
+    The value type must be specified statically, unlike a Python
+    Counter, which can accept arbitrary value types.
+
+    The value type must implement the `KeyElement` trait, as its values are
+    stored in the dictionary as keys. `KeyElement` includes
+    `Movable`, `Hashable`, and `EqualityComparable`.
+
+    Usage:
+
+    ```mojo
+    from collections import Counter
+    var c = Counter[String](["a", "a", "a", "b", "b", "c", "d", "c", "c"])
```

Comment:
I think you need to write `List(...)` instead.

Reply:
Unfortunatelly, you are right. Fixed: https://github.com/msaelices/mojo/commit/bda132997618e46081ed22f186a9dbfd23f1a016